documentation/src/test/java/org/hibernate/search/documentation/search/query/QueryDslIT.java

```diff
@@ -417,6 +418,49 @@ public void fetchSize() {
 		} );
 	}
 
+	@Test
+	public void resultTotal() {
+		OrmUtils.withinJPATransaction( entityManagerFactory, entityManager -> {
+			SearchSession searchSession = Search.session( entityManager );
+			// tag::fetching-resultTotal[]
+			SearchResult<Book> result = searchSession.search( Book.class )
+					.where( f -> f.matchAll() )
+					.fetch( 20 );
+
+			SearchResultTotal resultTotal = result.total(); // <1>
+			long totalHitCount = resultTotal.hitCount(); // <2>
+			long totalHitCountLowerBound = resultTotal.hitCountLowerBound(); // <3>
+			boolean hitCountExact = resultTotal.isHitCountExact(); // <4>
+			boolean hitCountLowerBound = resultTotal.isHitCountLowerBound(); // <5>
+			// end::fetching-resultTotal[]
+
+			assertThat( totalHitCount ).isEqualTo( 4 );
+			assertThat( totalHitCountLowerBound ).isEqualTo( 4 );
+			assertThat( hitCountExact ).isEqualTo( true );
+			assertThat( hitCountLowerBound ).isEqualTo( false );
+		} );
+	}
+
+	@Test
+	public void resultTotal_totalHitsThreshold() {
+		OrmUtils.withinJPATransaction( entityManagerFactory, entityManager -> {
+			SearchSession searchSession = Search.session( entityManager );
+			// tag::fetching-totalHitsThreshold[]
+			SearchResult<Book> result = searchSession.search( Book.class )
+					.where( f -> f.matchAll() )
+					.totalHitsThreshold( 2 ) // <1>
```

Comment:
Maybe use a more realistic value here, e.g. `1000`? It's not a big deal if we don't reach the threshold in this test: we already have tests elsewhere.